pkg/compute/store/boltdb/counter.go

```diff
@@ -0,0 +1,50 @@
+package boltdb
+
+import (
+	"github.com/bacalhau-project/bacalhau/pkg/compute/store"
+	"go.uber.org/atomic"
+)
+
+type StateCounter struct {
+	data map[store.ExecutionState]*atomic.Uint64
+}
```

Comment:
I am assuming it is safe to use a map instead of syncMap since the counter is only mutated within blot transactions, right?

Reply:
More specifically, because it's only mutated within the single allowed writer transaction.